boards/arm/beagle_play_cc1352/board_antenna.c

```diff
@@ -0,0 +1,117 @@
+/* SPDX-License-Identifier: Apache-2.0
+ *
+ * Copyright (c) 2021 Florin Stancu
+ * Copyright (c) 2021 Jason Kridner, BeagleBoard.org Foundation
+ *
+ */
+
+/*
+ * Implements the RF driver callback to configure the on-board antenna
+ * switch.
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/init.h>
+
+#include <ti/drivers/rf/RF.h>
+#include <driverlib/gpio.h>
+#include <driverlib/ioc.h>
+#include <driverlib/rom.h>
+
+/* DIOs for RF antenna paths */
+#define BOARD_RF_HIGH_PA   29		/* TODO: pull from DT */
+#define BOARD_RF_SUB1GHZ   30		/* TODO: pull from DT */
+
+static void board_cc13xx_rf_callback(RF_Handle client, RF_GlobalEvent events,
+		void *arg);
+
+
+const RFCC26XX_HWAttrsV2 RFCC26XX_hwAttrs = {
+	.hwiPriority        = INT_PRI_LEVEL7,
+	.swiPriority        = 0,
+	.xoscHfAlwaysNeeded = true,
+	/* RF driver callback for custom antenna switching */
+	.globalCallback = board_cc13xx_rf_callback,
+	/* Subscribe to events */
+	.globalEventMask = (RF_GlobalEventRadioSetup |
+			RF_GlobalEventRadioPowerDown),
+};
+
+/**
+ * Antenna switch GPIO init routine.
+ */
+static int board_antenna_init(const struct device *dev)
+{
+	ARG_UNUSED(dev);
+
```

Comment:
I guess this was just changed to `void` somewhat recently

Reply:
@Ayush1325 Ping, since you are actively working on BeagleConnect freedom and Beagleplay, can you fixup the minor changes required in this PR?

Reply:
@vaishnavachath This PR totally slipped by radar. I already do have a PR for beagleplay on-board cc1352: https://github.com/zephyrproject-rtos/zephyr/pull/64718. Other than caparray delta (which I am planning to add after https://github.com/zephyrproject-rtos/zephyr/pull/65048, my PR also contains everything rebased on upstream zephyr

Reply:
No problem @Ayush1325 , can you add the Caparray delta to https://github.com/zephyrproject-rtos/zephyr/pull/65048 (new commit) rather than creating a new PR for that alone? and please confirm if this can be closed as duplicate.

Reply:
@vaishnavachath Yes, I will add caparray to https://github.com/zephyrproject-rtos/zephyr/pull/65048